pkgs/development/tools/sourcetrail/default.nix

```diff
@@ -1,77 +1,207 @@
+  #
+  # Considering the problems with the above layouts, we go with the third
+  # option, a slight variation of the AppImage layout:
+  #
+  # nixpkgs
+  # ├── bin/
+  # │   └── sourcetrail* (wrapper for opt/sourcetrail/bin/sourcetrail)
+  # └── opt/sourcetrail/
+  #     ├── bin/
+  #     │   └── sourcetrail*
+  #     └── share/
+  #         └── data/
+  #
+  # Upstream install script:
+  # https://github.com/CoatiSoftware/Sourcetrail/blob/master/setup/Linux/createPackages.sh
+  installPhase = ''
+    mkdir -p $out/opt/sourcetrail/bin
+    cp app/Sourcetrail $out/opt/sourcetrail/bin/sourcetrail
+    cp app/sourcetrail_indexer $out/opt/sourcetrail/bin/sourcetrail_indexer
+
+    desktop-file-install --dir=$out/share/applications \
+      --set-key Exec --set-value $out/bin/sourcetrail \
+      ../setup/Linux/data/sourcetrail.desktop
+
+    mkdir -p $out/share/mime/packages
+    cp ../setup/Linux/data/sourcetrail-mime.xml $out/share/mime/packages/
+
+    mkdir -p $out/opt/sourcetrail/share
+    cp -R ../bin/app/data $out/opt/sourcetrail/share
+    cp -R ../bin/app/user/projects $out/opt/sourcetrail/share/data/fallback
+    rm $out/opt/sourcetrail/share/data/*_template.xml
+    rm -r $out/opt/sourcetrail/share/data/install
+
+    for size in 48 64 128 256 512; do
+      mkdir -p $out/share/icons/hicolor/''${size}x''${size}/apps/
+      convert app/data/gui/icon/logo_1024_1024.png -resize ''${size}x''${size} \
+        $out/share/icons/hicolor/''${size}x''${size}/apps/sourcetrail.png
+    done
+
+    mkdir -p $out/bin
+    makeQtWrapper $out/opt/sourcetrail/bin/sourcetrail $out/bin/sourcetrail \
+      --prefix PATH : ${stdenv.lib.makeBinPath binPath}
+  '';
 
-    [ -d "\$HOME" ] && setup_config
-    export PATH="\$PATH:${python3}/bin"
-    exec "$out/opt/Sourcetrail.sh" "\$@"
-    EOF
+  checkPhase = ''
+    rm -rf ../bin/app/data/{python,java/lib}
+    ln -s $out/opt/sourcetrail/share/data/python ../bin/app/data/python
+    ln -s $out/opt/sourcetrail/share/data/java/lib ../bin/app/data/java/lib
+
+    pushd test
+    # shorten PATH to prevent build failures
+    wrapQtApp ./Sourcetrail_test \
+      --set PATH "" \
+      --prefix PATH : ${stdenv.lib.makeBinPath testBinPath} \
+      --set MAVEN_OPTS "-Dmaven.repo.local=$TMPDIR/m2repo"
+    ./Sourcetrail_test
+    popd
+
+    rm ../bin/app/data/{python,java/lib}
+  '';
 
-    chmod +x $out/bin/sourcetrail
+  # This has to be done manually in the installPhase because the actual binary
+  # lives in $out/opt/sourcetrail/bin, which isn't covered by wrapQtAppsHook
+  dontWrapQtApps = true;
 
-    runHook postInstall
-  '';
+  # FIXME: some test cases are disabled in the patch phase
+  doCheck = true;
 
   meta = with stdenv.lib; {
     homepage = "https://www.sourcetrail.com";
     description = "A cross-platform source explorer for C/C++ and Java";
-    platforms = [ "x86_64-linux" ];
-    license = licenses.unfree;
-    maintainers = with maintainers; [ midchildan ];
```

Comment:
Did you remove the maintainers list on purpose?

Reply:
Oops. I rewrote this package from scratch and forgot to re-add it.